### PR TITLE
Refactor storage shape to reduce lookups required for third parties.

### PR DESCRIPTION
**Pull request by jonathanKingston**
You can review it this time @princiya as I haven't had time to check this over much. Also this is your code and I want you to still be aware of it.

### CODE REVIEW

File: js/store.js [hunk 1]
```diff
@@ -45,17 +45,17 @@ const store = {
...
 
-  outputWebsite(hostname, website, firstPartyHostname) {
+  outputWebsite(hostname, website) {
     const output = {
       hostname: hostname,
       favicon: website.faviconUrl || '',
-      firstPartyHostname: firstPartyHostname || false,
+      firstPartyHostnames: website.firstPartyHostnames || false,
       firstParty: false
     };
```
Comment (princiya) [inline js/store.js]: @jonathanKingston can we add `thirdParties: []` here and leave out the below `else` statement? In that way, the `output` object's signature would be complete.

File: js/store.js [hunk 2]
```diff
@@ -45,17 +45,17 @@ const store = {
     return await browser.storage.local.set({ websites });
   },
 
-  outputWebsite(hostname, website, firstPartyHostname) {
+  outputWebsite(hostname, website) {
     const output = {
       hostname: hostname,
       favicon: website.faviconUrl || '',
-      firstPartyHostname: firstPartyHostname || false,
+      firstPartyHostnames: website.firstPartyHostnames || false,
```
Comment (princiya) [inline js/store.js]: Why is `firstPartyHostnames` an array now?

Comment (jonathanKingston) [inline js/store.js]: Because a third party can belong to multiple firsts.

File: js/store.js (hunk 1 above)
Comment (jonathanKingston) [inline js/store.js]: The intention was the key wasn't present but either is fine.

File: js/store.js (hunk 2 above)
Comment (jonathanKingston) [inline js/store.js]: The same thing you complained wasn't always accurate ;)

**Pull request merged**